compiler/modules/nback/debug.bal

```diff
@@ -0,0 +1,230 @@
+            }
+            else if child.startPos < declPos && child.endPos >= registerScope.endPos {
+                return self.addRegisterScope(registerScope, declPos, child);
+            }
+        }
+        Scope newScope = self.createScope(declPos, registerScope.endPos, parent, []);
+        if addIndex is () {
+            parent.childScopes.push(newScope);
+        }
+        else {
+            Scope[] newChildScopes = parent.childScopes.slice(0, addIndex);
+            newChildScopes.push(newScope, ...parent.childScopes.slice(addIndex));
+            parent.childScopes = newChildScopes;
+        }
+        return newScope;
+    }
+
+    function currentPos() returns bir:Position {
+        return <bir:Position> self.currentPosition;
+    }
+
+    function clearDebugLocation(llvm:Builder builder) {
+        // in debugFull case every instruction must have a debug location
+        if !self.debugFull {
+            builder.setCurrentDebugLocation(());
+            self.currentDebugLocation = ();
+        }
+    }
+
+    function setCurrentPosition(llvm:Builder builder, bir:Position pos) {
+        self.currentPosition = pos;
+        if self.debugFull {
+            self.currentDebugLocation = self.debugLocation(pos);
+            builder.setCurrentDebugLocation(self.currentDebugLocation);
+        }
+        else {
+            self.currentDebugLocation = ();
+        }
+    }
+
+    function useDebugLocation(llvm:Builder builder, DebugLocationUsage usage) {
+        // In the debugFull case, there is no need to do anything for DEBUG_USAGE_ERROR_CONSTRUCT
+        // because the full location will have been set earlier.
+        if usage == DEBUG_USAGE_ERROR_CONSTRUCT && self.debugFull {
+            return;
+        }
+        DILocation loc;
+        if usage == DEBUG_USAGE_ERROR_CONSTRUCT {
+            DILocation? noLineLoc = self.noLineLocation;
+            if noLineLoc == () {
+                loc = self.debugLocation(0);
+                self.noLineLocation = loc;
+            }
+            else {
+                loc = noLineLoc;
+            }
+        }
+        else {
+            if self.currentDebugLocation == () {
+                self.currentDebugLocation = self.debugLocation(self.currentPos());
+            }
+            loc = <DILocation>self.currentDebugLocation;
+        }
+        builder.setCurrentDebugLocation(loc);
+    }
+
+    function debugLocation(bir:Position pos) returns DILocation {
+        var [line, column] = self.scaffold.lineColumn(pos);
+        return self.diBuilder.createDebugLocation(line, column, diScope(pos, self.rootScope));
+    }
+}
+
+function diScope(bir:Position pos, Scope parent) returns DIScope {
+    foreach var child in parent.childScopes {
+        if child.startPos <= pos && child.endPos >= pos {
+            return diScope(pos, child);
+        }
+    }
+    return parent.diScope;
+}
+
+function semTypeToDWARF(t:SemType semType) returns DWARFType {
```

Comment:
Ballerina style is to capitalize just first letter of acronym (e.g. Xml, Http), so Dwarf not DWARF.

Reply:
fixed